### PR TITLE
Make videos open in modals

### PR DESCRIPTION
**Pull request by richardxia**
## Overview

This changes the video links on the home page to open up a modal with the video in an embedded YouTube player, allowing users to watch the video without opening the link in a separate window.

## Screenshots

Screen Shot 2020-11-22 at 5 17 50 PM
Screen Shot 2020-11-22 at 5 18 07 PM

## Detailed changes

https://github.com/ShelterTechSF/sheltertech.org/commit/36116cabe797efd5f6764a0c3fd0fb04a629dd14 and https://github.com/ShelterTechSF/sheltertech.org/commit/1e6910b2cfed0e1fa349f1e8b4dd87b13fb93900 are smaller changes to the `VideoHeader` component and the `VideoSpotlightBlock` that change the `playButtonLink` props to `playButtonOnClick`. This is needed because instead of containing external links, this should now just trigger a React state change to open the modal.

https://github.com/ShelterTechSF/sheltertech.org/commit/3b8008761879839d224c12709161f16068bb3f10 modifies the `<Modal>` component to add a new `noBezel` variant. When `noBezel` is set, then we don't render the white container box with the padding, the close button, and the rounded corners. I initially tried putting the video embed in the normal modal container, but it looked pretty weird, so instead I decided that it would probably look the best with no "bezel" at all.

https://github.com/ShelterTechSF/sheltertech.org/commit/3f6734221d0695e5de063971d528fd6a382cc8c6 implements the actual `YouTubeEmbed` component, which is just a lightweight wrapper around the embed `<iframe>` that you get from YouTube. The main thing that we add on top is a wrapper that provides an aspect-ratio-preserving style on top.

Finally, https://github.com/ShelterTechSF/sheltertech.org/commit/3332f4d98c07b716876dd1cf4ad36c92bbe6ab19 actually ties everything together by adding the new video modal onto the home page.

### CODE REVIEW

File: src/components/grid-aware/VideoHeader/VideoHeader.stories.js [hunk 1]
```diff
@@ -28,5 +28,4 @@ DefaultVideoHeader.args = {
     { text: "Donate", internalLink: "/new/donate" },
     { text: "Volunteer", internalLink: "/new/volunteer" },
```
Comment (candywang) [inline src/components/grid-aware/VideoHeader/VideoHeader.stories.js:29]: I understand this is the `stories.js` file and won't be used in production but did we want to update the `internalLink` paths to `"/donate"` and `"/volunteer"` respectively even if it doesn't make a difference?

**Review by candywang — APPROVED** [review]
LGTM!

**Review by davidagustin — APPROVED** [review]
All questions asked through personal conversation.

File: src/components/grid-aware/VideoHeader/VideoHeader.stories.js (hunk 1 above)
Comment (richardxia) [reply src/components/grid-aware/VideoHeader/VideoHeader.stories.js:29]: I don't think it's really necessary, and if we were to do something here, I'd probably want to just replace these with completely dummied out URLs, just to emphasize that the actual URLs in the links is not important in the Storybook story.

**Comment by richardxia** [comment]
Thanks for quickly reviewing this!

File: src/components/grid-aware/VideoHeader/VideoHeader.stories.js (hunk 1 above)
Comment (candywang) [reply src/components/grid-aware/VideoHeader/VideoHeader.stories.js:29]: Okay! Thanks for the reference.

**Pull request merged**